### PR TITLE
Initial playground component

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Related to https://github.com/bpmn-io/internal-docs/issues/617
Depends on https://github.com/bpmn-io/form-js/pull/329

This includes
* Prepare initial project setup (cf. https://github.com/camunda/form-playground/pull/1/commits/84519632fc1e7afdd52fc1f49ca2eac60d0c8baf)
* Prepare demo (cf. https://github.com/camunda/form-playground/pull/1/commits/00588a96fe1122bab94ee50dc5c7146063259d65 + https://github.com/camunda/form-playground/pull/1/commits/c9ee43ca30f251529876a026175de7e33fde37f0)
* Add layout context (cf. https://github.com/camunda/form-playground/pull/1/commits/957d370ab18f9b9ed47aea2a7c0d81a9c02ff9d6)
* Add collapsible panel component (cf. https://github.com/camunda/form-playground/pull/1/commits/65f221f2de57e2a3df24e1d5f2bb9fea88b44a5a)
* Add collapsed preview component (cf. https://github.com/camunda/form-playground/pull/1/commits/7d2d14317350e789f2f41954103448c5bdaafae8)
* Add `CamundaFormPlayground` component (cf. https://github.com/camunda/form-playground/pull/1/commits/f47360d8600cf57a41a67a451ac57caeb962b861)
* Update README (cf. https://github.com/camunda/form-playground/pull/1/commits/299debfbe2a4837f0be673e8bd427e78d35e2b44)

Demo: https://camunda-form-playground.netlify.app/
Some more contextual help: https://docs.google.com/presentation/d/1Uac0zANg3d3-66Xbry1L7GgiCONstGgSX9Fgcz4ayEc/edit?usp=sharing

image

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
> ### ✅ Deploy Preview for _camunda-form-playground_ ready!
> Name	Link
> 🔨 Latest commit	[4aec29f](https://github.com/camunda/form-playground/commit/4aec29f01ca760034ccea36e902a9f66ff6d908c)
> 🔍 Latest deploy log	https://app.netlify.com/sites/camunda-form-playground/deploys/63107e4ac0f9df00081caedb
> 😎 Deploy Preview	https://deploy-preview-1--camunda-form-playground.netlify.app
> 📱 Preview on mobile	
> Toggle QR Code...
> _To edit notification comments on pull requests, go to your [Netlify site settings](https://app.netlify.com/sites/camunda-form-playground/settings/deploys#deploy-notifications)._

This was not intended, but it is pretty cool indeed...

**Comment by pinussilvestrus** [comment]
I will some more context for reviewing this in the next hour of code (cf. https://github.com/bpmn-io/internal-docs/issues/635). Feel free to already have a look.

**Comment by pinussilvestrus** [comment]
[CI is currently failing](https://github.com/camunda/form-playground/actions/runs/3088227493/jobs/4994457472#step:7:63) due to an a11y issue. See also https://github.com/bpmn-io/form-js/pull/335

**Comment by pinussilvestrus** [comment]
Resolved by integrating `form-js@0.9.1`.

**Comment by barmac** [comment]
Is this ready for review?

**Comment by pinussilvestrus** [comment]
Yes! With giving the lightning talk, this should be ready to have a look 👍 I added the slides and sketches in the description.

**Comment by barmac** [comment]
From a user perspective, I'd like us to prevent flickering when one switches mode:

https://user-images.githubusercontent.com/28307541/191933612-14f839a6-f286-466f-bec5-a4e497af7626.mov

**Comment by pinussilvestrus** [comment]
I will check whether this is something related to the demo or also visible in the Desktop Modeler) 👍

File: README.md [hunk 1]
```diff
@@ -6,7 +6,173 @@ Camunda Form Playground.
...
+
+```javascript
+// open all
+formPlayground.open();
+
+// open specific
+formPlayground.open([ 'form-preview' ]);
+```
+
+### `CamundaFormPlayground#collapse(containers: String[]) => void`
```
Comment (barmac) [inline README.md:90]:
```suggestion
### `CamundaFormPlayground#collapse(containers?: String[]) => void`
```

If it's optional, we should mark it as optional (also in the type definitions if applies).

Comment (barmac) [reply README.md:90]: BTW the TS way would be to use lowercase `string`.

File: package.json [hunk 2]
```diff
@@ -13,9 +13,12 @@
     "build": "rollup -c",
     "build:watch": "rollup -cw",
     "lint": "eslint src",
-    "start": "cd example && npm start",
+    "start": "run-p build:watch start:example",
```
Comment (barmac) [inline package.json:16]: I had a problem with this as I had to manually `cd example && npm ci` in order to run the start script. We could consider either using workspaces, or postinstall/prepare hook in the root, or lift the dev deps to the root.

**Comment by pinussilvestrus** [comment]
> I will check whether this is something related to the demo or also visible in the Desktop Modeler) 👍

It's also present over there, so let's handle this carefully (cf. our meeting on Monday).

File: README.md (hunk 1 above)
Comment (pinussilvestrus) [reply README.md]: Adjusted it via https://github.com/camunda/form-playground/pull/1/commits/2e84b44abea82a0c5ec06672ced203c34e69072c

File: package.json (hunk 2 above)
Comment (pinussilvestrus) [reply package.json:16]: I decided to use workspaces via https://github.com/camunda/form-playground/pull/1/commits/0e41d37302985e46fa049f32de3c667ab5136b1d

**Comment by pinussilvestrus** [comment]
Regarding the flickering problem: After adding a keyboard shortcut via https://github.com/camunda/form-playground/pull/1/commits/1ce43b83b20cb6f82bc7f707db9523ecb5358137 to the demo I was able to figure out this is caused by the panel headers of _Form input_ (look at the bottom near the engine profile select) and _Form output_ (look at the top below the _Form preview_ header).

Seems like the headers are rendered quicker than the rest after the transition from `display: none` to `display: flex`.

Kapture 2022-09-29 at 08 42 25

**Comment by barmac** [comment]
Excalidraw sketch: https://excalidraw.com/#json=oqApfumbu2W5vI-O_EeXo,YNbjs24eMjb6NLfFK3G6ug

**Comment by pinussilvestrus** [comment]
**Root cause for the flickering**

We handle displaying the collapsible content in two places (cf. [sketch above](https://github.com/camunda/form-playground/pull/1#issuecomment-1261962719)), which leads to two individual frames:

Kapture 2022-09-29 at 10 46 03

We need to make sure to handle the display in _one single transition_.

**Comment by pinussilvestrus** [comment]
@barmac I ended up with handling the `open` states globally in the parent `PlaygroundComponent`. Rationale: These friends are anyway really depending on the other's open states, so it just makes sense to handle it in a central place.

This should now behave smooth:

Kapture 2022-09-29 at 14 26 46

Cf. https://github.com/camunda/form-playground/pull/1/commits/b237c36957fa0d230ab423260a8d92ea5cdc72e9

Let me know if there is anything else we should consider. If not, let's merge it!

**Review by barmac — APPROVED** [review]

**Pull request merged**